sdk/resourcemanager/Proto.Client/compute/VirtualMachineScaleSetRollingUpgradeOperations.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.ResourceManager.Compute;
+using Azure.ResourceManager.Core;
+
+namespace Proto.Compute
+{
+    /// <summary> A class representing the operations that can be performed over a specific VirtualMachineScaleSetRollingUpgrade. </summary>
+    public partial class VirtualMachineScaleSetRollingUpgradeOperations : SingletonOperationsBase<ResourceGroupResourceIdentifier, VirtualMachineScaleSetRollingUpgrade>
+    {
+        /// <summary> Initializes a new instance of VirtualMachineScaleSetRollingUpgradeOperations for mocking. </summary>
+        protected VirtualMachineScaleSetRollingUpgradeOperations()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="VirtualMachineOperations"/> class.
+        /// </summary>
+        /// <param name="operation"> The client parameters to use in these operations. </param>
+        /// <param name="id"> The identifier of the resource that is the target of operations. </param>
+        protected internal VirtualMachineScaleSetRollingUpgradeOperations(OperationsBase operation, ResourceGroupResourceIdentifier id)
+            : base(operation, id)
+        {
+        }
+
+        public static readonly ResourceType ResourceType = "Proto.Compute/VirtualMachineScaleSet/VirtualMachineScaleSetRollingUpgradeOperations";
+
+        protected override ResourceType ValidResourceType => ResourceType;
+
+        private VirtualMachineScaleSetRollingUpgradesOperations Operations => new ComputeManagementClient(
+            BaseUri,
+            Id.SubscriptionId,
+            Credential,
+            ClientOptions.Convert<ComputeManagementClientOptions>()).VirtualMachineScaleSetRollingUpgrades;
+
+        // Note: Singleton may have different operations such as GET/PUT/PATCH/POST or a combination of these
+        // Individual methods will be generated as they are declared
+        public ArmResponse<VirtualMachineScaleSetRollingUpgrade> Get(CancellationToken cancellationToken = default)
+        {
+            return new PhArmResponse<VirtualMachineScaleSetRollingUpgrade, Azure.ResourceManager.Compute.Models.RollingUpgradeStatusInfo>(
+                Operations.GetLatest(Id.ResourceGroupName, "", cancellationToken),
+                v => new VirtualMachineScaleSetRollingUpgrade(this, new RollingUpgradeStatusInfo(v)));
+        }
+
+        public Task<ArmResponse<VirtualMachineScaleSetRollingUpgrade>> GetAsync(CancellationToken cancellationToken = default)
+        {
+            throw new NotImplementedException();
```

Comment:
I assume this is just skipped for time and not something we can't implement?

Reply:
Right. I was gonna come back and finish it with scenario tests.